Replace CALM Hub UI's visualiser with the one from calm-explorer

pmerrison opened the pull request:
## Description
This PR replaces the cytoscape based visualiser in calm-hub-ui with the one from experimental/calm-vizualiser.  It includes re-themeing it to match the existing colour scheme, and brings along the node details panel.

This PR does not bring over any of the other UI elements, we can think about what else we might want for a follow up PR

## Type of Change
<!-- Check the type that applies to your PR -->
- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [ ] 🎨 Code style/formatting changes
- [ ] ♻️ Refactoring (no functional changes)
- [ ] ⚡ Performance improvements
- [ ] ✅ Test additions or updates
- [ ] 🔧 Chore (maintenance, dependencies, CI, etc.)

## Affected Components
<!-- Check all that apply -->
- [ ] CLI (`cli/`)
- [ ] Shared (`shared/`)
- [ ] CALM Widgets (`calm-widgets/`)
- [ ] CALM Hub (`calm-hub/`)
- [X] CALM Hub UI (`calm-hub-ui/`)
- [ ] Documentation (`docs/`)
- [ ] VS Code Extension (`calm-plugins/vscode/`)
- [ ] Dependencies
- [ ] CI/CD

## Commit Message Format ✅
<!-- 
Make sure your commit messages follow the conventional commit format:
type(scope): description

Scope is optional but recommended - you can use ./commit-helper.sh to get suggestions!

Examples:
- feat(cli): add new validation command
- fix(shared): resolve schema parsing issue
- docs: update installation guide
- chore(deps): bump typescript to 5.8.3

This helps with our automated versioning and changelog generation!
Note: Only commits with (cli) scope will trigger CLI releases.
-->

## Testing
<!-- Describe how you tested your changes -->
- [X] I have tested my changes locally
- [X] I have added/updated unit tests
- [X] All existing tests pass

## Checklist
- [X] My commits follow the [conventional commit format](https://www.conventionalcommits.org/)
- [ ] I have updated documentation if necessary
- [X] I have added tests for my changes (if applicable)
- [X] My changes follow the project's coding standards

aidanm3341 commented on calm-hub-ui/src/visualizer/components/reactflow/theme.ts:
```diff
@@ -0,0 +1,97 @@
+ * Matches calm-hub-ui's DaisyUI light theme
+ */
+
+export const THEME = {
```
I think some of these are duplicated, like the primary and accent colors which should already be defined somewhere else. Would be good to define all these constants in the same way and same place.

aamanrebello commented:
Nice! It would be great if the code in individual files was simpler and modularised though - currently there is a lot of stuff in each file.

pmerrison commented on calm-hub-ui/src/visualizer/components/reactflow/ArchitectureGraph.tsx:
```diff
@@ -0,0 +1,789 @@
+import { THEME } from './theme';
+
+interface ArchitectureGraphProps {
+  jsonData: any;
```
This should be fixed in the latest commit

aidanm3341 commented on calm-hub-ui/src/visualizer/components/reactflow/ControlsPanel.tsx:
```diff
@@ -0,0 +1,453 @@
+
+    const controlEntries = Object.entries(controls);
+
+    return (
```
this chunk of JSX is 410 lines long and _deeply_ nested 😅 

you never want to see JSX that ends with

```jsx
                                                    </span>
                                                </div>
                                            </div>
                                        )}
                                    </div>
                                </div>
                            )}
                        </div>
                    ))}
                </div>
            </div>
        </div>
    );
}
```

very much not ideal for reading and understanding, would be good to break it down

aidanm3341 commented on calm-hub-ui/src/visualizer/components/reactflow/CustomNode.tsx:
```diff
@@ -0,0 +1,352 @@
+interface RiskItem {
+    id?: string;
+    name?: string;
+    description?: string;
```
do these fields need to be optional? The less undefined values we need to work with, the better

pmerrison commented:
@aidanm3341 - I think I've addressed all your comments now, here's a summary of what I've changed

 **Quick Wins**

  1. Function syntax conversion - Converted 4 components from arrow
  functions to function syntax (ArchitectureGraph, CustomNode, FloatingEdge,
   SystemGroupNode)
  2. CSS variable removal - Removed duplicate CSS variables from index.css,
  kept hardcoded hex values matching theme/colors.ts

  **Testing**

  3. floatingEdges.ts tests - Added 9 tests for getEdgeParams() function

  **Feature Restoration**

  4. Drawer title - Restored title state and display showing file name or
  CALM Hub metadata

  **Major Refactoring**

  5. Removed options relationship logic - Deleted
  expandOptionsRelationships() function and its tests (simplifies
  architecture-only visualization)
  6. calmTransformer.ts refactoring - Split into:
    - nodeParser.ts - Node parsing with identifyContainerNodes(),
  parseNodes()
    - relationshipParser.ts - Relationship parsing with
  extractFlowTransitions(), parseRelationships()
    - Main file reduced from 478 to 228 lines, now a clean orchestrator
  7. ControlsPanel.tsx refactoring - Split into:
    - controls-panel/ControlsPanelHeader.tsx
    - controls-panel/ControlCard.tsx
    - controls-panel/ControlRequirementItem.tsx
    - controls-panel/AIGFMappingSection.tsx
    - Main file reduced from 453 to 71 lines
  8. FlowsPanel.tsx refactoring - Split into:
    - flows-panel/FlowsPanelHeader.tsx
    - flows-panel/FlowCard.tsx
    - flows-panel/FlowTransitionItem.tsx
    - flows-panel/AIGFGovernanceSection.tsx
    - Main file reduced from 312 to 61 lines

  **Documentation**

  9. Optional fields documentation - Added clarifying comments explaining
  why fields are optional in CustomNode.tsx and FloatingEdge.tsx

  **Verification**

  10. ADR functionality - All 213 tests pass, including ADR-related tests

  New files created: 13
  Files modified: 10

aamanrebello commented on calm-hub-ui/src/visualizer/components/drawer/Drawer.tsx:
```diff
@@ -1,52 +1,167 @@
+ * Extract the unique-id from a CALM node or relationship
+ */
+function extractId(item: CalmNodeSchema | CalmRelationshipSchema): string {
+    return item?.['unique-id'] || '';
```
I feel like we can use a better default value than empty string, something more obvious maybe like `ID_PLACEHOLDER`

aamanrebello commented on calm-hub-ui/src/visualizer/components/reactflow/theme.ts:
Should this be in the themes folder?

aamanrebello commented on calm-hub-ui/src/visualizer/components/reactflow/MetadataPanel.tsx:
```diff
@@ -0,0 +1,276 @@
+                        alignItems: 'center',
+                        justifyContent: 'center',
+                        padding: '6px',
+                        border: 'none',
```
These templates could be simplified possibly.
